test/gtest/ucs/test_string.cc

```diff
@@ -41,6 +41,24 @@ UCS_TEST_F(test_string_buffer, appendf) {
     ucs_string_buffer_cleanup(&strb);
 }
 
+UCS_TEST_F(test_string_buffer, append_long) {
+    ucs_string_buffer_t strb;
+    std::string str;
+
+    str.resize(100);
+    std::fill(str.begin(), str.end(), 'e');
+
+    ucs_string_buffer_init(&strb);
+
+    ucs_string_buffer_appendf(&strb, "%s", str.c_str());
```

Comment:
```
const unsigned max_iters = 2;
...
std::string exp_str;
for (unsigned i = 0; i < max_iters; i++) {
    ucs_string_buffer_appendf(&strb, "%s", str.c_str());
    exp_str += str;
    EXPECT_EQ(exp_str.c_str(), std::string(ucs_string_buffer_cstr(&strb)));
}
```
wdyt about this approach instead of two append+expect_eq? it could be flexible to change and test more iterations if needed